extensions/shared/src/main/java/app/revanced/extension/youtube/sponsorblock/SegmentPlaybackController.java

```diff
@@ -545,6 +543,54 @@ private static void setSegmentCurrentlyPlaying(@Nullable SponsorSegment segment)
         SponsorBlockViewController.showSkipSegmentButton(segment);
     }
 
+    /**
+     * Handles changes in player control visibility and manages the skip segment button accordingly.
+     *
+     * <p>This method is called whenever the visibility state of the player controls changes.
+     * If auto-hide is enabled and there is a currently playing sponsor segment, it will show
+     * the skip segment button when the controls are visible and schedule recursive checks
+     * to hide the button after a defined duration.</p>
+     *
+     * @param visible {@code true} if player controls are visible; {@code false} otherwise
+     */
+    public static void onPlayerControlsVisibilityChanged(boolean visible) {
+        if (!Settings.SB_AUTO_HIDE_SKIP_BUTTON.get() || segmentCurrentlyPlaying == null) {
+            return;
+        }
+
+        if (visible) {
+            SponsorBlockViewController.showSkipSegmentButton(segmentCurrentlyPlaying);
+            skipSegmentButtonEndTime = System.currentTimeMillis() + DURATION_TO_SHOW_SKIP_BUTTON;
+            checkPlayerControlsVisibilityRecursive(segmentCurrentlyPlaying);
+        }
+    }
+
+    /**
+     * Recursively checks whether the skip segment button should remain visible or be hidden.
+     *
+     * <p>This method continues checking at a fixed interval (500 milliseconds) if the button
+     * should be hidden. The recursion stops if the current segment changes or the duration
+     * to show the button has expired.</p>
+     *
+     * @param segment the sponsor segment associated with the current check
+     */
+    private static void checkPlayerControlsVisibilityRecursive(SponsorSegment segment) {
+        // Stop recursion if the current segment has changed
+        if (segment != segmentCurrentlyPlaying) {
+            return;
+        }
+
+        // Continue recursion if the button's visibility duration has not expired
+        if (skipSegmentButtonEndTime > System.currentTimeMillis()) {
+            Utils.runOnMainThreadDelayed(() -> checkPlayerControlsVisibilityRecursive(segment), 500);
```

Comment:
In general, the SponsorBlock segments are checked every 1000ms, so there is no problem even if the delay is set to 1000ms.